File: 08-03-2025_Longest_Palindrome_in_a_String.java
static String longestPalindrome(String s) {
    int n = s.length();
    int begin = 0, end = 0, len = 0;
    String ans = "";

    for (int i = 0; i < n; i++) {
        for (int j = 0; j <= 1; j++) {
            int x = i;  // Case for odd length
            int y = i + j; // Case for even length

            while (x >= 0 && y < n && s.charAt(x) == s.charAt(y)) {
                int currLen = y - x + 1;
                if (currLen > len) {
                    begin = x;
                    end = y;
                    len = currLen;
                }
                x--;
                y++;
            }
        }
    }

    return s.substring(begin, end + 1);
}

File: 08-03-2025_Longest_Palindrome_in_a_String.cpp
string longestPalindrome(string &s) {
    // code here
    int n = s.size();
    int begin, end, len = 0;
    string ans;
    
    for(int i = 0; i < n; i++){
        for(int j = 0; j <= 1; j++){
            int x = i;  //Case of odd length
            int y = i+j; //Case of even length
            
            while(x >= 0 && y < n && s[x] == s[y]){
                int currLen = y - x + 1;
                if(currLen > len){
                    begin = x;
                    end = y;
                    len = currLen;
                }
                x--;
                y++;
            }
        }
    }
    
    while(begin <= end){
        ans += s[begin];
        begin++;
    }
    
    return ans;
}
